sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp/test-aot-compat.ps1

```diff
@@ -23,12 +24,24 @@ foreach ($line in $($publishOutput -split "`r`n"))
 }
 
 Write-Host "Actual warning count is:", $actualWarningCount
-$expectedWarningCount = 7
+$expectedWarningCount = 16
```

Comment:
The 4 from the JsonConverters are the only that are expected

Reply:
After taking your changes, I can confirm this is down to 4. All are related to JsonConverters. :)
Is there a plan for these outstanding warnings?

Reply:
The outstanding warnings have to be baselined in tests, I can link the discussion that led to this decision if you're interested in more information, but to briefly summarize, `DynamicData` and `MutableJsonDocument` are not AOT compatible at all partially because of the custom converters. If we suppress the warnings on the converters and only annotate the actual class, there are ways that non-AOT compatible code could get missed by the trimmer (it's unlikely - but still possible). So, to be safe, we opted to leave the warnings on these classes.

Since the test uses the <TrimmerRootAssembly> tag, the trimmer is very liberal in the code it considers part of this assembly. Because of this, `DynamicData` and `MutableJsonDocument` get pulled in because there is a constructor in `RequestContent` that has a `DynamicData` parameter, even though (I think) the constructor-in-question is not actually used anywhere in your library. This means that as long as your library isn't using `DynamicData` anywhere, a customer using `Azure.Monitor.OpenTelemetry.Exporter` won't see these warnings on their end.

I hope I explained this okay, but happy to answer other questions if needed.